packages/monaco/src/browser/monaco-quick-input-service.ts

```diff
@@ -580,6 +580,14 @@ class MonacoQuickPick<T extends QuickPickItem> extends MonacoQuickInput implemen
         return this.wrapped.items.map(item => QuickPickSeparator.is(item) ? item : item.item);
     }
 
+    get buttons(): ReadonlyArray<IQuickInputButton> {
```

Comment:
The typing here is not right: there is no "buttons" property on `QuickPick`. Returning monaco-specific types here is not appropriate. `QuickPick` is a wrapper around `IQuickPick`.

Reply:
So how should we handle this?
The current implementation in the [quick-open-main](https://github.com/eclipse-theia/theia/blob/b15b639fa3b64a9518efdf3e82ea9098915ba145/packages/plugin-ext/src/main/browser/quick-open-main.ts#L272) does not do any type checks at all. It simply iterates through the "params" (of type `TransferQuickPick`) list and assigns them as property of the `input` element.

So basically we would have to make sure that all properties of the `TransferQuickPick` are present on the wrapper and properly forwared to the underyling monaco specific element.

On a sidenote: There is also a wrapper for `MonacoQuickInput` which seems currently unused. That`s why the buttons works when using `QuickInput` because in that case we use the monaco type directly.
We should probably also wrap that correctly ore remove this class altogether

Reply:
Add the buttons types `readonly QuickInputButton[]` to the `QuickPick` interface and fix the typing by appropriately casting/converting.
Which wraper is unused? Can you explain, please?

Reply:
@tsmaeder 

Unfortunatley the `QuickInputButton` and Monacos `IQuickInputButton` interface are not compatible.
They use different URI types for the `iconPath`:
 - QuickInputButton: import from 'vscode-uri';
-  IQuickInputButton: import  from '@theia/monaco-editor/lib/vs/base/common/uri`

They difference is the `UriComponent` definition which has partial optional properties in the monaco variant
```
export interface UriComponents {
	scheme: string;
	authority?: string;
	path?: string;
	query?: string;
	fragment?: string;
}
``` 

but has no optional properties in the vscode-uri variant:
export interface UriComponents {
    scheme: string;
    authority: string;
    path: string;
    query: string;
    fragment: string;
}
```

Reply:
Can't you just cast the relevant structures? The values are not changed by the cast anyway.

Reply:
👍🏼 done